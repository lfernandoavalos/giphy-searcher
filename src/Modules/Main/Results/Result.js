import React from 'react';
import PropTypes from 'prop-types';

import { Card, CardMedia, CardTitle } from 'react-toolbox/lib/card';


import styles from './../styles.css';

const getPreviewUrl = url => url.replace('media2', 'i');

const Result = ({ title, user, images }) => (
  <div className={styles.result}>
    <Card raised={false}>
      <CardTitle
        avatar={user.avatar_url}
        title={user.username}
        subtitle={`Twitter: ${user.twitter}`}
      />
      <CardMedia
        aspectRatio="wide"
        image={getPreviewUrl(images.original.url)}
      />
      <CardTitle
        title={title}
      />
    </Card>
  </div>
);

Result.defaultProps = {
  user: {
    username: 'Anonymous',
    avatar_url: '#',
    twitter: 'anonymous',
  },
};

Result.propTypes = {
  title: PropTypes.string.isRequired,
  user: PropTypes.shape({
    username: PropTypes.string.isRequired,
    avatar_url: PropTypes.string.isRequired,
    twitter: PropTypes.string.isRequired,
  }),
  images: PropTypes.shape({
    original: PropTypes.shape({
      url: PropTypes.string.isRequired,
    }),
  }).isRequired,
};

export default Result;
