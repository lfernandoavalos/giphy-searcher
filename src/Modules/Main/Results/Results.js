import React from 'react';
import PropTypes from 'prop-types';

import Result from './Result';

const Results = ({ results }) => (
  <div>
    {results.map(result => (
      <Result key={result.id} result={result} />
    ))}
  </div>
);

Results.propTypes = {
  results: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    user: PropTypes.shape({
      avatar_url: PropTypes.string.isRequired,
      twitter: PropTypes.string.isRequired,
    }).isRequired,
    images: PropTypes.shape({
      original: PropTypes.shape({
        url: PropTypes.string.isRequired,
      }),
    }),
  })).isRequired,
};

export default Results;
