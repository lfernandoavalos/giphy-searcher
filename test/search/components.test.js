import React from 'react';
import renderer from 'react-test-renderer';

import { TRENDING, SEARCH_RESULTS } from './__mock__';

import Results from './../../src/Modules/Main/Results/Results';
import Result from './../../src/Modules/Main/Results/Result';

it('renders gif results correclty', () => {
  const tree = renderer
    .create(<Results results={TRENDING} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders gif without user result correclty', () => {
  const tree = renderer
    .create(<Result {...SEARCH_RESULTS[0]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders gif with user result correclty', () => {
  const tree = renderer
    .create(<Result {...TRENDING[0]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
