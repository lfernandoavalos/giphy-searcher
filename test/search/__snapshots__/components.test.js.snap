// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders gif results correclty 1`] = `
<div>
  <div
    className="result"
  >
    <div
      className=""
      data-react-toolbox="card"
    >
      <div
        className=""
      >
        <div
          className=""
          data-react-toolbox="avatar"
        >
          <img
            alt=""
            className={undefined}
            src="https://media.giphy.com/avatars/msnbc/kaVAvQ8Xdzvx.jpg"
          />
        </div>
        <div>
          <h5
            className={undefined}
          >
            msnbc
          </h5>
          <p
            className={undefined}
          >
            Twitter: @MSNBC
          </p>
        </div>
      </div>
      <div
        className="undefined"
        style={
          Object {
            "backgroundColor": undefined,
            "backgroundImage": "url('https://media0.giphy.com/media/fx86vKE0pXj1qkpKZF/giphy.gif')",
          }
        }
      >
        <div
          className=""
        />
      </div>
      <div
        className="undefined"
      >
        <div>
          <h5
            className={undefined}
          >
            chris matthews lol GIF by MSNBC
          </h5>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders gif with user result correclty 1`] = `
<div
  className="result"
>
  <div
    className=""
    data-react-toolbox="card"
  >
    <div
      className=""
    >
      <div
        className=""
        data-react-toolbox="avatar"
      >
        <img
          alt=""
          className={undefined}
          src="https://media.giphy.com/avatars/msnbc/kaVAvQ8Xdzvx.jpg"
        />
      </div>
      <div>
        <h5
          className={undefined}
        >
          msnbc
        </h5>
        <p
          className={undefined}
        >
          Twitter: @MSNBC
        </p>
      </div>
    </div>
    <div
      className="undefined"
      style={
        Object {
          "backgroundColor": undefined,
          "backgroundImage": "url('https://media0.giphy.com/media/fx86vKE0pXj1qkpKZF/giphy.gif')",
        }
      }
    >
      <div
        className=""
      />
    </div>
    <div
      className="undefined"
    >
      <div>
        <h5
          className={undefined}
        >
          chris matthews lol GIF by MSNBC
        </h5>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders gif without user result correclty 1`] = `
<div
  className="result"
>
  <div
    className=""
    data-react-toolbox="card"
  >
    <div
      className=""
    >
      <div
        className=""
        data-react-toolbox="avatar"
      >
        <img
          alt=""
          className={undefined}
          src="#"
        />
      </div>
      <div>
        <h5
          className={undefined}
        >
          Anonymous
        </h5>
        <p
          className={undefined}
        >
          Twitter: anonymous
        </p>
      </div>
    </div>
    <div
      className="undefined"
      style={
        Object {
          "backgroundColor": undefined,
          "backgroundImage": "url('https://i.giphy.com/media/Dz62ImmWRzAkw/giphy.gif')",
        }
      }
    >
      <div
        className=""
      />
    </div>
    <div
      className="undefined"
    >
      <div>
        <h5
          className={undefined}
        >
          family guy quagmire GIF
        </h5>
      </div>
    </div>
  </div>
</div>
`;
